.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,24 @@
+name: Unit Tests
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin' # See 'Supported distributions' for available options
+          java-version: '11'
```

Comment:
I have upgraded this repo to use Java 17 yesterday, so let's use 17 here as well